lib/Doctrine/ORM/Cache/NullCacheItemPool.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Cache;
+
+use Closure;
+use Doctrine\Common\Cache\Psr6\CacheItem;
+use Generator;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+class NullCacheItemPool implements CacheItemPoolInterface
+{
+    /** @var Closure|false|null */
+    private static $createCacheItem;
+
+    public function __construct()
+    {
+        self::$createCacheItem ?? self::$createCacheItem = Closure::bind(
+            static function ($key) {
+                return new CacheItem($key, null, false);
+            },
+            null,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
```

Comment:
You can use as many native type declarations as possible in this class, please do so.